src/organizers/block_organizer.cpp

```diff
@@ -75,8 +75,8 @@ bool block_organizer::start()
 bool block_organizer::stop()
 {
     validator_.stop();
+    downloader_subscriber_->relay(error::service_stopped, null_hash, (size_t)0);
     downloader_subscriber_->stop();
```

Comment:
By not stopping before broadcasting this allows resubscription after the service_stopped message. The purpose of the service_stopped message is to give every subscription the opportunity to clear itself (which it must). But this race results in a subscriber never seeing the service_stopped message, and therefore a shutdown deadlock.